core/src/main/java/org/springframework/security/authorization/method/Jsr250AuthorizationManager.java

```diff
@@ -81,7 +81,11 @@ public void setRolePrefix(String rolePrefix) {
 	@Override
 	public AuthorizationDecision check(Supplier<Authentication> authentication, MethodInvocation methodInvocation) {
 		AuthorizationManager<MethodInvocation> delegate = this.registry.getManager(methodInvocation);
-		return delegate.check(authentication, methodInvocation);
+		AuthorizationDecision decision = delegate.check(authentication, methodInvocation);
+		if (decision == null) {
+			return null;
+		}
+		return new AuthorizationManagerAuthorizationDecision(decision.isGranted(), delegate);
```

Comment:
I think `AuthorityAuthorizationManager` might be a more reasonable place since it could return the granted authorities needed. It could return `AuthorityAuthorizationDecision`, for example.

Returning a service component as a member may not be as helpful as you are thinking since it will not typically have any getters.

Reply:
Yeah, I think it was an oversight because I did not see the `AuthorityAuthorizationManager`